core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -400,16 +438,35 @@ public DefaultMuleEvent(MuleMessage message,
 
     protected boolean resolveEventSynchronicity()
     {
-        boolean syncProcessingStrategy = false;
+        return transacted
+               || isSynchronousProcessingStrategy()
+               || exchangePattern.hasResponse() && !isNonBlockingProcessingStrategy()
+               || message.getProperty(MuleProperties.MULE_FORCE_SYNC_PROPERTY,
+                                      PropertyScope.INBOUND, Boolean.FALSE);
+    }
+
+    private boolean isSynchronousProcessingStrategy()
+    {
         if (flowConstruct != null && flowConstruct instanceof ProcessingDescriptor)
```

Comment:
This condition can be simplified to return (flowConstruct instanceof ProcessingDescriptor) && ((ProcessingDescriptor) flowConstruct).isSynchronous()